src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,203 @@
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
+    /// A <see cref="Utf8JsonStreamReaderConverter{T}"/> to allow read JSON into <see cref="LockFileTargetLibrary"/>
+    /// </summary>
+    internal class Utf8JsonStreamLockFileTargetLibraryConverter : Utf8JsonStreamReaderConverter<LockFileTargetLibrary>
+    {
+        private static readonly byte[] Utf8Type = Encoding.UTF8.GetBytes("type");
+        private static readonly byte[] Utf8Framework = Encoding.UTF8.GetBytes("framework");
+        private static readonly byte[] Utf8Dependencies = Encoding.UTF8.GetBytes("dependencies");
+        private static readonly byte[] Utf8FrameworkAssemblies = Encoding.UTF8.GetBytes("frameworkAssemblies");
+        private static readonly byte[] Utf8Runtime = Encoding.UTF8.GetBytes("runtime");
+        private static readonly byte[] Utf8Compile = Encoding.UTF8.GetBytes("compile");
+        private static readonly byte[] Utf8Resource = Encoding.UTF8.GetBytes("resource");
+        private static readonly byte[] Utf8Native = Encoding.UTF8.GetBytes("native");
+        private static readonly byte[] Utf8Build = Encoding.UTF8.GetBytes("build");
+        private static readonly byte[] Utf8BuildMultiTargeting = Encoding.UTF8.GetBytes("buildMultiTargeting");
+        private static readonly byte[] Utf8ContentFiles = Encoding.UTF8.GetBytes("contentFiles");
+        private static readonly byte[] Utf8RuntimeTargets = Encoding.UTF8.GetBytes("runtimeTargets");
+        private static readonly byte[] Utf8Tools = Encoding.UTF8.GetBytes("tools");
+        private static readonly byte[] Utf8Embed = Encoding.UTF8.GetBytes("embed");
+        private static readonly byte[] Utf8FrameworkReferences = Encoding.UTF8.GetBytes("frameworkReferences");
+
+        private IList<PackageDependency> ReadPackageDependencyList(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                return new List<PackageDependency>(0);
+            }
+
+            var packageDependencies = new List<PackageDependency>();
+            while (reader.Read() && reader.TokenType == JsonTokenType.PropertyName)
+            {
+                string propertyName = reader.GetString();
+                string versionString = reader.ReadNextTokenAsString();
+
+                packageDependencies.Add(new PackageDependency(
+                    propertyName,
+                    versionString == null ? null : VersionRange.Parse(versionString)));
+            }
+            return packageDependencies;
+        }
+
+        public override LockFileTargetLibrary Read(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException("Expected PropertyName, found " + reader.TokenType);
+            }
+
+            var lockFileTargetLibrary = new LockFileTargetLibrary();
```

Comment:
I image these are commonly referenced throughout an assets file, can we dedupe these to avoid parsing the same data over and over again?